### PR TITLE
Add missing code customization for Choreo functions

### PR DESCRIPTION
**Pull request by YasasRangika**
### Goals
To reduce the customization of master branch done on top of Choreo for supporting the organization concept.

### Approach
This PR adds and refactors the following changes:
- Changes from mini hackathon 2 
- pass organization directly instead getting from CarbonContext (mini hackathon 3)
- Changes from mini hackathon 4

https://docs.google.com/spreadsheets/d/18GA4odAnRz2cB0Ris3RL_62_yDfE3Jb52j5LP5BI2OI/edit#gid=0

### CODE REVIEW

**Review by tharikaGitHub — APPROVED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/APIMappingUtil.java [hunk 1]
```diff
@@ -167,8 +167,8 @@ public static APIDTO fromAPItoDTO(API model, String tenantDomain) throws APIMana
         Set<org.wso2.carbon.apimgt.api.model.Tier> apiTiers = model.getAvailableTiers();
         List<APITiersDTO> tiersToReturn = new ArrayList<>();
         int tenantId = 0;
-        if (!StringUtils.isBlank(tenantDomain)) {
-            tenantId = APIUtil.getTenantIdFromTenantDomain(tenantDomain);
+        if (!StringUtils.isBlank(organization)) {
+            tenantId = APIUtil.getTenantIdFromTenantDomain(organization);
```
Comment (tharikaGitHub) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/APIMappingUtil.java:171]: This should be changed to getInternalOrganizationId(organization);

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/APIMappingUtil.java [hunk 2]
```diff
@@ -320,8 +320,8 @@ public static APIDTO fromAPItoDTO(APIProduct model, String tenantDomain) throws
         List<APITiersDTO> tiersToReturn = new ArrayList<>();
 
         int tenantId = 0;
-        if (!StringUtils.isBlank(tenantDomain)) {
-            tenantId = APIUtil.getTenantIdFromTenantDomain(tenantDomain);
+        if (!StringUtils.isBlank(organization)) {
+            tenantId = APIUtil.getTenantIdFromTenantDomain(organization);
```
Comment (tharikaGitHub) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/APIMappingUtil.java:324]: Same here on getting the tenant ID

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/APIMappingUtil.java (hunk 1 above)
Comment (YasasRangika) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/APIMappingUtil.java]: fixed in https://github.com/wso2/carbon-apimgt/pull/10553/commits/448f1494c616bef5b514c816f3a29c6dda74a465

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/APIMappingUtil.java (hunk 2 above)
Comment (YasasRangika) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/APIMappingUtil.java]: fixed in https://github.com/wso2/carbon-apimgt/pull/10553/commits/448f1494c616bef5b514c816f3a29c6dda74a465

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 3]
```diff
@@ -2665,13 +2665,13 @@ public static APIStore getExternalAPIStore(String apiStoreName, int tenantId) th
...
-    public static Map<String, Tier> getTiers(int tierType, String tenantDomain) throws APIManagementException {
+    public static Map<String, Tier> getTiers(int tierType, String organization) throws APIManagementException {
 
         boolean isTenantFlowStarted = false;
         try {
             PrivilegedCarbonContext.startTenantFlow();
             isTenantFlowStarted = true;
-            PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain, true);
+            PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(organization, true);
             int tenantId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:2675]: this won't work for choreo

**Review by chamilaadhi — CHANGES_REQUESTED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java (hunk 3 above)
Comment (chamilaadhi) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java:2675]: I think u don't hv to start a tenant flow here. remove them. to get the tenantId, use getInternalOrganizationId(organization); method

**Review by chamilaadhi — CHANGES_REQUESTED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 4]
```diff
@@ -6463,15 +6463,15 @@ public static String getMountedPath(RegistryContext registryContext, String path
...
+    public static Map<String, String> getDomainMappings(String organization, String appType)
             throws APIManagementException {
 
         Map<String, String> domains = new HashMap<String, String>();
         String resourcePath;
         try {
             Registry registry = ServiceReferenceHolder.getInstance().getRegistryService().
                     getGovernanceSystemRegistry();
-            resourcePath = APIConstants.API_DOMAIN_MAPPINGS.replace(APIConstants.API_DOMAIN_MAPPING_TENANT_ID_IDENTIFIER, tenantDomain);
+            resourcePath = APIConstants.API_DOMAIN_MAPPINGS.replace(APIConstants.API_DOMAIN_MAPPING_TENANT_ID_IDENTIFIER, organization);
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: This won't work in choreo setup.  add this before passing the domain

    if (StringUtils.isNotEmpty(organization)) {
        int tenantId = APIUtil.getInternalOrganizationId(organization);
        // To handle choreo scenario. due to keymanagers are not per organization atm. using ST
        if (tenantId == MultitenantConstants.SUPER_TENANT_ID) {
            organization = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
        }
    }

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java (hunk 3 above)
Comment (YasasRangika) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: fixed in https://github.com/wso2/carbon-apimgt/pull/10553/commits/ac2ae3789c1ef92290e91e89f7a74571a7962869

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java (hunk 4 above)
Comment (YasasRangika) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: fixed in https://github.com/wso2/carbon-apimgt/pull/10553/commits/92236ff1b77766202a3d3cb139bfe7943a1b81aa

**Review by chamilaadhi — APPROVED** [review]

**Review by chamilaadhi — APPROVED** [review]

**Pull request merged**